DemoApp/DemoAppCoordinator.swift

```diff
@@ -34,33 +32,22 @@ final class DemoAppCoordinator: NSObject {
             showLogin(animated: false)
         }
     }
-}
 
-// MARK: - UNUserNotificationCenterDelegate
-
-extension DemoAppCoordinator: UNUserNotificationCenterDelegate {
-    func userNotificationCenter(
-        _ center: UNUserNotificationCenter,
-        didReceive response: UNNotificationResponse,
-        withCompletionHandler completionHandler: @escaping () -> Void
-    ) {
-        defer {
-            completionHandler()
-        }
-
-        guard let notificationInfo = try? ChatPushNotificationInfo(content: response.notification.request.content) else {
-            return
-        }
+    func handlePushNotificationResponse() {
+        pushNotifications.onNotificationResponse = { [weak self] response in
+            guard case UNNotificationDefaultActionIdentifier = response.actionIdentifier else {
+                return
+            }
 
-        guard let cid = notificationInfo.cid else {
-            return
-        }
+            guard let self = self else { return }
```

Comment:
I would do that as the first thing in closure and move checking `actionIdentifier` to the next guard OR event have a single guard for all statements

Reply:
Done!